### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by mosajjal** [mosajjal/sniproxy#52] [Announcements]
This is a major overhaul to `sniproxy` that introduces many breaking changes. in Version 2.x, the only way to configure `sniproxy` is through a YAML configuration file. please refer to the sample configuration file included or use  the following command to generate a default config file:

`./sniproxy --defaultconfig` 

notable features and changes:

- DoH support
- SOCKS5 support for proxy (including UDP)
- IP ACL 
- Manual override ACL
- [REMOVED] http reverse proxy support
- Performance improvements
- IPv6 support improvements
- GeoIP ACL logic enhancement

special thanks to @holygrolli  and  @prooshani for helping with the testing and contributing to the project.

**Full Changelog**: https://github.com/mosajjal/sniproxy/compare/v1.0.0...v2.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mosajjal/sniproxy/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by yaser-k** [comment]
thank you for this assume project.  I hope you consider supporting geoip  and geosite. it will be a great feature.
https://github.com/maxmind
https://github.com/v2fly/domain-list-community

**Comment by mosajjal** [comment]
geoip is supported through an ACL [here](https://github.com/mosajjal/sniproxy/blob/5f8bed8d524bf77c7564ca36c571d662e4d8ddf4/config.defaults.yaml#L50)

**Reply by yaser-k** [reply]
yes I notice that. I suggest it for outbound rules. replacement the domain.csv. geosite make it very easier to select what service we want to relay through proxy.

**Reply by mosajjal** [reply]
ah I see what you mean. feel free to open a new suggestion and let's see if the feature is in popular demand. if it is, I'm happy to write it :)
